pkg/injector/patch.go

```diff
@@ -54,22 +55,28 @@ func (wh *mutatingWebhook) createPatch(pod *corev1.Pod, req *admissionv1.Admissi
 	// Create volume for envoy TLS secret
 	pod.Spec.Volumes = append(pod.Spec.Volumes, getVolumeSpec(envoyBootstrapConfigName)...)
 
-	// Build outbound port exclusion list
-	podOutboundPortExclusionList, _ := wh.getPortExclusionListForPod(pod, namespace, outboundPortExclusionListAnnotation)
-	globalOutboundPortExclusionList := wh.configurator.GetOutboundPortExclusionList()
-	outboundPortExclusionList := mergePortExclusionLists(podOutboundPortExclusionList, globalOutboundPortExclusionList)
-
-	// Build inbound port exclusion list
-	podInboundPortExclusionList, _ := wh.getPortExclusionListForPod(pod, namespace, inboundPortExclusionListAnnotation)
-	globalInboundPortExclusionList := wh.configurator.GetInboundPortExclusionList()
-	inboundPortExclusionList := mergePortExclusionLists(podInboundPortExclusionList, globalInboundPortExclusionList)
-
-	// Add the Init Container
-	initContainer := getInitContainerSpec(constants.InitContainerName, wh.configurator, wh.configurator.GetOutboundIPRangeExclusionList(), outboundPortExclusionList, inboundPortExclusionList, wh.configurator.IsPrivilegedInitContainer())
-	pod.Spec.InitContainers = append(pod.Spec.InitContainers, initContainer)
+	// On Windows we cannot use init containers to program HNS because it requires elevated privileges
+	// As a result we assume that the HNS redirection policies are already programmed via a CNI plugin.
+	// Skip adding the init container and only patch the pod spec with sidecar container.
+	podOS := pod.Spec.NodeSelector["kubernetes.io/os"]
+	if !strings.EqualFold(podOS, "windows") {
```

Comment:
nit: define "windows" as a constant in `constants.go` if this is going to be used in other files as well.